Mathlib/LinearAlgebra/Eigenspace/Basic.lean

```diff
@@ -578,3 +578,54 @@ lemma map_add_of_iInf_genEigenspace_ne_bot_of_commute [NoZeroSMulDivisors R M]
 end End
 
 end Module
+
+section SpectrumDiagonal
+
+variable {R : Type*} [Field R] {n : Type*} [DecidableEq n][Fintype n]
+
+open Module.End
+
+/--  Standard basis vectors are eigenvectors of any associated diagonal linear operator. -/
+lemma Matrix.hasEigenvector_toLin'_diagonal (d : n → R) (i : n) :
```

Comment:
This is the first mention of `Matrix` in this file; perhaps that means this content would be better in a new `EigenSpace/Matrix.lean` file?

Reply:
Cool. Let me confer with Jireh about how best to do this.

Reply:
@eric-wieser : Should I create this new file and move the result as part of this PR, or should I handle this in another PR? (I'm still trying to get the hang of this!)

Reply:
Yes, just make the new file and move this result into it as part of this PR. Generally, I would recommend basing what should constitute a separate PR in terms of semantically / mathematically distinct content (although there can be exceptions to this).

On a separate note: the `b : Basis _ _ _` from the other comment above probably should also apply here.